src/main/java/core/basesyntax/strategy/operation/ReturnOperationHandlerImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class ReturnOperationHandlerImpl implements OperationHandler {
+    @Override
+    public void updateAmount(FruitTransaction fruitTransaction) {
+        String currentFruit = fruitTransaction.getFruit();
+        int currentAmount = Storage.fruitStorage.getOrDefault(currentFruit, 0);
+        int transactionAmount = fruitTransaction.getQuantity();
+        int updatedAmount = currentAmount + transactionAmount;
+        if (updatedAmount < 0) {
+            throw new RuntimeException("Fruit quantity cannot be negative!");
+        }
```

Comment:
```suggestion
```
you add Amount it can't be negative

Reply:
I thought about receiving a negative number. Just extra check. But if it's redundant, I will delete it.